### PR TITLE
Torch Compile - New Op Support

### PR DESCRIPTION
**Pull request by cavusmustafa**
New op support for:
 - torch.export updates
 - benchmarking model support
 - chatglm2 support

### CODE REVIEW

File: src/frontends/pytorch/src/op/sort.cpp [hunk 1]
```diff
@@ -41,6 +41,36 @@ OutputVector translate_sort(const NodeContext& context) {
...
+    // aten.sort.default(Tensor self, int dim=-1, bool descending=False) -> (Tensor values, Tensor indices)
+    num_inputs_check(context, 1, 3);
+    const auto input_tensor = context.get_input(0);
+    bool descending = false;
+    int64_t dim = -1;
+
+    if (!context.input_is_none(1)) {
+        dim = context.const_input<int64_t>(1);
+    }
+    if (!context.input_is_none(1)) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/sort.cpp:54]:
```suggestion
    if (!context.input_is_none(2)) {
```

File: src/frontends/pytorch/src/op/sort.cpp [hunk 2]
```diff
@@ -41,6 +41,36 @@ OutputVector translate_sort(const NodeContext& context) {
...
+    auto shape = context.mark_node(std::make_shared<opset11::ShapeOf>(input_tensor));
+    auto k_values_node = context.mark_node(std::make_shared<opset11::Gather>(shape, dim_axis, zero_axis));
+    auto k_values = context.mark_node(std::make_shared<opset11::Squeeze>(k_values_node));
+    auto topk = context.mark_node(std::make_shared<opset11::TopK>(input_tensor,
+                                                                  k_values,
+                                                                  dim,
+                                                                  mode,
+                                                                  ov::op::TopKSortType::SORT_VALUES,
+                                                                  element::i64));
+    auto indices = context.mark_node(std::make_shared<ov::op::v0::Convert>(topk->output(1), element::i64));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/sort.cpp:70]: This part is similar between ts and fx, could you extract it in `common` function?

File: src/frontends/pytorch/src/op/var_mean.cpp [hunk 3]
```diff
@@ -145,6 +145,11 @@ OutputVector translate_var(const NodeContext& context) {
     return {res[0]};
 }
 
+OutputVector translate_var_correction_fx(const NodeContext& context) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var_mean.cpp:148]: This function is not used in `op_table.cpp`

File: tests/layer_tests/pytorch_tests/test_batch_norm.py [hunk 4]
```diff
@@ -62,4 +62,4 @@ def forward(self, x):
     @pytest.mark.precommit_torch_export
     def test_batch_norm(self, weights, bias, eps, train, running_stats, ie_device, precision, ir_version, kwargs_to_prepare_input):
         self._test(*self.create_model(weights, bias, eps, train, running_stats),
-                   ie_device, precision, ir_version, kwargs_to_prepare_input=kwargs_to_prepare_input, dynamic_shapes=False, use_mo_convert=False)
\ No newline at end of file
+                   ie_device, precision, ir_version, kwargs_to_prepare_input=kwargs_to_prepare_input, dynamic_shapes=False, use_mo_convert=False)
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_batch_norm.py:65]: Please revert this change

File: tests/layer_tests/pytorch_tests/test_flip.py [hunk 5]
```diff
@@ -41,4 +41,4 @@ def forward_out(self, x, y):
     @pytest.mark.parametrize("out", [skip_if_export(True), False])
     @pytest.mark.parametrize("dtype", ["float32", "float64", "int32", "int64", "uint8"])
     def test_flip(self, axis, out, dtype, ie_device, precision, ir_version):
-        self._test(*self.create_model(axis, out), ie_device, precision, ir_version, kwargs_to_prepare_input={"out": out, "dtype": dtype})
\ No newline at end of file
+        self._test(*self.create_model(axis, out), ie_device, precision, ir_version, kwargs_to_prepare_input={"out": out, "dtype": dtype})
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_flip.py:44]: Please remove this change

File: tests/layer_tests/pytorch_tests/test_hardswish.py
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_hardswish.py:1]: Since this op doesn't have any arguments, please add hardswish in `test_unary_ops.py` rather then separate file.

File: tests/layer_tests/pytorch_tests/test_hardtanh.py
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_hardtanh.py:1]: Could you also add tests for different types, like f64 and int if original pytorch op support it

File: tests/layer_tests/pytorch_tests/test_masked_fill.py [hunk 6]
```diff
@@ -67,7 +68,6 @@ def test_masked_fill(self, value, mask_fill, mask_dtype, input_dtype, inplace, i
     @pytest.mark.parametrize("mask_dtype", [np.uint8, np.int32])  # np.float32 incorrectly casted to bool
     @pytest.mark.parametrize("inplace", [True, False])
     @pytest.mark.nightly
-    @pytest.mark.precommit
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_masked_fill.py:70]: Why this is removed?

**Review by mvafin — CHANGES_REQUESTED** [review]

**Comment by mvafin** [comment]
Could you also add `aten.isfinite` since it is also streightforward and makes sense to test with a common test??

File: tests/layer_tests/pytorch_tests/test_roll.py [hunk 7]
```diff
@@ -18,12 +18,12 @@ class aten_roll(torch.nn.Module):
             def __init__(self, shifts, dim=None):
                 super(aten_roll, self).__init__()
                 self.dim = dim
-                self.shits = shifts
+                self.shifts = shifts
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_roll.py:21]: Could you also fix same problem here:
https://github.com/openvinotoolkit/openvino/blob/221847de5b24fcc47d82c8774f8f8a8c09d2df51/tests/layer_tests/py_frontend_tests/test_torch_decoder.py#L482

File: src/frontends/pytorch/src/op/bitwise.cpp [hunk 8]
```diff
@@ -7,6 +7,8 @@
 #include "openvino/op/bitwise_not.hpp"
 #include "openvino/op/bitwise_or.hpp"
 #include "openvino/op/bitwise_xor.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/logical_not.hpp"
```
[the remark this answers is not in the document]
Comment (cavusmustafa) [reply src/frontends/pytorch/src/op/bitwise.cpp]: This was part of a translation update for ChatGLM2 and should not be part of this PR. I removed these lines.

File: src/frontends/pytorch/src/op/isinf.cpp
[the remark this answers is not in the document]
Comment (cavusmustafa) [reply src/frontends/pytorch/src/op/isinf.cpp:1]: Yes, I replaced isinf and isnan translations with 1_to_1_match

File: src/frontends/pytorch/src/op/isnan.cpp
[the remark this answers is not in the document]
Comment (cavusmustafa) [reply src/frontends/pytorch/src/op/isnan.cpp:1]: explained above

File: src/frontends/pytorch/src/op_table.cpp [hunk 9]
```diff
@@ -885,7 +910,7 @@ const std::map<std::string, CreatorFunction> get_supported_ops_fx() {
         {"prim::PythonOp", op::translate_pythonop},
         {"prim::requires_grad", op::return_false_scalar},
         {"prim::type", op::skip_node},  // Used with prim::device, pass PtFrameworkNode.
```
[the remark this answers is not in the document]
Comment (cavusmustafa) [reply src/frontends/pytorch/src/op_table.cpp]: We can remove them with this PR and I updated accordingly. Only two torchvision:: ops are left since they are used.

File: tests/layer_tests/pytorch_tests/test_select_scatter.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+            def forward(self, x):
+                return torch.select_scatter(x, self.src, self.dim, self.index);
+
+
+        ref_net = None
+
+        return aten_select_scatter(src, dim, index), ref_net, "aten::select_scatter"
+
+    import torch
```
[the remark this answers is not in the document]
Comment (cavusmustafa) [reply tests/layer_tests/pytorch_tests/test_select_scatter.py]: It was used for torch.ones(2) to generate parameter. I moved this import to the top of the file.

File: tests/layer_tests/pytorch_tests/test_log.py
[the remark this answers is not in the document]
Comment (cavusmustafa) [reply tests/layer_tests/pytorch_tests/test_log.py:1]: I removed it as well but not sure if it's right way to do it. I will revert the related commit if needed.

File: src/frontends/pytorch/src/op/var_mean.cpp [hunk 11]
```diff
@@ -160,4 +160,4 @@ OutputVector translate_std_mean(const NodeContext& context) {
 }  // namespace op
 }  // namespace pytorch
 }  // namespace frontend
-}  // namespace ov
\ No newline at end of file
+}  // namespace ov
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/var_mean.cpp:163]: Please remove this change

**Review by mvafin — CHANGES_REQUESTED** [review]

**Comment by cavusmustafa** [comment]
> Could you also add `aten.isfinite` since it is also streightforward and makes sense to test with a common test??

Pushed the changes for 'aten.isfinite' support

File: src/frontends/pytorch/src/op/sort.cpp [hunk 12]
```diff
@@ -39,6 +26,43 @@ OutputVector translate_sort(const NodeContext& context) {
...
+
+    if (!context.input_is_none(1)) {
+        dim = context.const_input<int64_t>(1);
+    }
+    if (!context.input_is_none(2)) {
+        descending = context.const_input<bool>(2);
+    }
+
+    auto topk_outputs = translate_sort_common(context, stable, dim, descending);
+    auto indices = context.mark_node(std::make_shared<ov::op::v0::Convert>(topk_outputs[1], element::i64));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/sort.cpp:64]: Do you really need to convert topk indices to i64 if you have TopK with `element::i64` created?

**Review by mvafin — CHANGES_REQUESTED** [review]
Please also fix codestyle

File: tests/layer_tests/pytorch_tests/test_unary_ops.py [hunk 13]
```diff
@@ -241,12 +245,14 @@ def test_unary_op_out(self, op_type, dtype, ie_device, precision, ir_version):
...
     @pytest.mark.nightly
     @pytest.mark.precommit
+    @pytest.mark.precommit_fx_backend
     @pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
     @pytest.mark.parametrize("op_type",
                              [
                                  "aten::relu6",
                                  "aten::selu",
                                  "aten::silu",
+                                 "aten::hardswish"
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_unary_ops.py:255]:
```suggestion
                                 "aten::hardswish",
```

File: src/frontends/pytorch/src/op_table.cpp [hunk 14]
```diff
@@ -848,11 +866,11 @@ const std::map<std::string, CreatorFunction> get_supported_ops_fx() {
         {"aten.ne.Scalar", op::translate_1to1_match_2_inputs_align_types<opset10::NotEqual>},
         {"aten.ne.Tensor", op::translate_1to1_match_2_inputs_align_types<opset10::NotEqual>},
         {"aten.neg.default", op::translate_neg},
-        {"aten.new_full.default", op::translate_new_full_fx},
-        {"aten.new_ones.default", op::translate_new_ones_fx},
+        {"aten.new_full.default", op::translate_new_full},
+        {"aten.new_ones.default", op::translate_new_ones},
```
Comment (mvafin) [inline src/frontends/pytorch/src/op_table.cpp:870]: Remove these changes

File: src/frontends/pytorch/src/op_table.cpp [hunk 15]
```diff
@@ -848,11 +866,11 @@ const std::map<std::string, CreatorFunction> get_supported_ops_fx() {
...
         {"aten.new_zeros.default", op::translate_new_zeros_fx},
-        {"aten.ones.default", op::translate_ones_fx},
-        {"aten.ones.names", op::translate_ones_fx},
+        {"aten.ones.default", op::translate_ones},
+        {"aten.ones.names", op::translate_ones},
```
Comment (mvafin) [inline src/frontends/pytorch/src/op_table.cpp:873]: Same here

File: src/frontends/pytorch/src/op/log.cpp [hunk 16]
```diff
@@ -30,7 +30,7 @@ OutputVector translate_log_sigmoid(const NodeContext& context) {
                                 op_vector.size());
     auto sigmoid = op_vector[0];
     auto log = context.mark_node(std::make_shared<v0::Log>(sigmoid));
-    return {log};
+    return {context.mark_node(make_list_construct(log->outputs()))};
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/log.cpp:33]: Please create `translate_log_sigmoid_fx`, because this func is already used in ts part. 

At this point you could already create a decorator for it, similar to `optional_out`

File: src/frontends/pytorch/src/op/div.cpp [hunk 17]
```diff
@@ -90,6 +90,17 @@ OutputVector translate_div_fx(const NodeContext& context) {
     return translate_div_common(context, x, y, rounding_mode, false);
 };
 
+OutputVector translate_div_fx_(const NodeContext& context) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/div.cpp:93]: Where is this used?

Comment (suryasidd) [reply src/frontends/pytorch/src/op/div.cpp:93]: Missed committing the changes in op_table.cpp. Thanks for catching it maxim

**Comment by suryasidd** [comment]
build_jenkins

**Comment by mvafin** [comment]
build_jenkins

**Review by mvafin — APPROVED** [review]
